### ISSUE TITLE
Can not connect to chrome cuz the Webdriver doesn't support ver 115

### ISSUE THREAD

**Issue by PanaceaWen** [ultrafunkamsterdam/undetected-chromedriver#1437]
`D:\anaconda3\lib\site-packages\undetected_chromedriver\__init__.py in __init__(self, options, user_data_dir, driver_executable_path, browser_executable_path, port, enable_cdp_events, service_args, service_creationflags, desired_capabilities, advanced_elements, service_log_path, keep_alive, log_level, headless, version_main, patcher_force_close, suppress_welcome, use_subprocess, debug, no_sandbox, user_multi_procs, **kw)
    451             service = None
    452 
--> 453         super(Chrome, self).__init__(
    454             executable_path=self.patcher.executable_path,
    455             port=port,

D:\anaconda3\lib\site-packages\selenium\webdriver\chrome\webdriver.py in __init__(self, executable_path, port, options, service_args, desired_capabilities, service_log_path, chrome_options, service, keep_alive)
     68             service = Service(executable_path, port, service_args, service_log_path)
     69 
---> 70         super(WebDriver, self).__init__(DesiredCapabilities.CHROME['browserName'], "goog",
     71                                         port, options,
     72                                         service_args, desired_capabilities,

D:\anaconda3\lib\site-packages\selenium\webdriver\chromium\webdriver.py in __init__(self, browser_name, vendor_prefix, port, options, service_args, desired_capabilities, service_log_path, service, keep_alive)
     91 
     92         try:
---> 93             RemoteWebDriver.__init__(
     94                 self,
     95                 command_executor=ChromiumRemoteConnection(

D:\anaconda3\lib\site-packages\selenium\webdriver\remote\webdriver.py in __init__(self, command_executor, desired_capabilities, browser_profile, proxy, keep_alive, file_detector, options)
    267         self.file_detector = file_detector or LocalFileDetector()
    268         self.start_client()
--> 269         self.start_session(capabilities, browser_profile)
    270 
    271     def __repr__(self):

D:\anaconda3\lib\site-packages\undetected_chromedriver\__init__.py in start_session(self, capabilities, browser_profile)
    714         if not capabilities:
    715             capabilities = self.options.to_capabilities()
--> 716         super(selenium.webdriver.chrome.webdriver.WebDriver, self).start_session(
    717             capabilities, browser_profile
    718         )

D:\anaconda3\lib\site-packages\selenium\webdriver\remote\webdriver.py in start_session(self, capabilities, browser_profile)
    358         parameters = {"capabilities": w3c_caps,
    359                       "desiredCapabilities": capabilities}
--> 360         response = self.execute(Command.NEW_SESSION, parameters)
    361         if 'sessionId' not in response:
    362             response = response['value']

D:\anaconda3\lib\site-packages\selenium\webdriver\remote\webdriver.py in execute(self, driver_command, params)
    423         response = self.command_executor.execute(driver_command, params)
    424         if response:
--> 425             self.error_handler.check_response(response)
    426             response['value'] = self._unwrap_value(
    427                 response.get('value', None))

D:\anaconda3\lib\site-packages\selenium\webdriver\remote\errorhandler.py in check_response(self, response)
    245                 alert_text = value['alert'].get('text')
    246             raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here
--> 247         raise exception_class(message, screen, stacktrace)
    248 
    249     def _value_or_default(self, obj: Mapping[_KT, _VT], key: _KT, default: _VT) -> _VT:

WebDriverException: Message: unknown error: cannot connect to chrome at 127.0.0.1:4004
from session not created: This version of ChromeDriver only supports Chrome version 113
Current browser version is 115.0.5790.171
Stacktrace:
Backtrace:
	GetHandleVerifier [0x007D8893+48451]
	(No symbol) [0x0076B8A1]
	(No symbol) [0x00675058]
	(No symbol) [0x00695D9C]
	(No symbol) [0x00690151]
	(No symbol) [0x0068FF31]
	(No symbol) [0x006C0798]
	(No symbol) [0x006C047C]
	(No symbol) [0x006BA0B6]
	(No symbol) [0x00697E08]
	(No symbol) [0x00698F2D]
	GetHandleVerifier [0x00A38E3A+2540266]
	GetHandleVerifier [0x00A78959+2801161]
	GetHandleVerifier [0x00A7295C+2776588]
	GetHandleVerifier [0x00862280+612144]
	(No symbol) [0x00774F6C]
	(No symbol) [0x007711D8]
	(No symbol) [0x007712BB]
	(No symbol) [0x00764857]
	BaseThreadInitThunk [0x761800C9+25]
	RtlGetAppContainerNamedObjectPath [0x76FA7B1E+286]
	RtlGetAppContainerNamedObjectPath [0x76FA7AEE+238]`

After I manually updated my Google to version 115 today, the UC stopped working and throw this error. Anyone know how to fix it?

**Comment by jdholtz** [comment]
There is currently a problem with fetching v115 of the chromedriver (see #1419). You can try specifying `version_main=114` in the Chrome object as v115 is compatible with Chromedriver 114

**Comment by jdholtz** [comment]
> Do this method work for you I wonder?

It did, so the root cause of this issue might be something unrelated to the issue I linked to.

**Comment by PanaceaWen** [comment]
Ha I got it Bro. I tried to update the latest version 115 manually and replace the old one in my Anaconda root path. And it worked when I was using the selenium, failed when I tried to create a driver using UC(And throwing me same error: My chromedriver remains version 113).

Then I realized the UC has its own Chromedriver.exe and cuz the Google fuck changed the Chromedriver Updating Url so the Automatically-Patcher of the UC fucked up. And what we should do it to manually patch the Undetected-Chromedriver.exe. Accroding to this friend's solution(https://github.com/ultrafunkamsterdam/undetected-chromedriver/discussions/1429#discussioncomment-6635534), I successfully fix it, Thank you my Amigo 👍
